Mage.Sets/src/mage/cards/m/MasterOfCeremonies.java

```diff
@@ -0,0 +1,178 @@
+                return "";
+            }
+
+            // Draw is an option since we won't deck ourselves
+            if (aiDecidingPlayer.getLibrary().size() != 0) {
+                return "Secret";
+            }
+
+            // TODO: Not sure what a simple criteria for choosing between a 1/1 and a treasure token would be,
+            //       But the 1/1 is likely less useful than a treasure.
+            return "Friends";
+        });
+
+        // Go through each vote and apply effects
+        for (UUID opponentId : game.getOpponents(controller.getId())) {
+            Player opponent = game.getPlayer(opponentId);
+            if (opponent == null) {
+                continue;
+            }
+
+            // Iterate through each vote that opponent took, may have multiple votes from certain events
+            // E.g. Ballot Broker
+            List<String> opponentsVotes = vote.getVotes(opponentId);
+            for (String opponentsVote : opponentsVotes) {
+                switch (opponentsVote) {
+                    case "Money": // You and that player each create a Treasure token.
+                        Token treasureOpponent = new TreasureToken();
+                        treasureOpponent.putOntoBattlefield(1, game, source, opponentId);
+
+                        Token treasureController = new TreasureToken();
+                        treasureController.putOntoBattlefield(1, game, source, controller.getId());
+
+                        break;
+                    case "Friends": // You and that player each create a 1/1 green and white Citizen creature token.
+                        Token citizenOpponent = new CitizenGreenWhiteToken();
+                        citizenOpponent.putOntoBattlefield(1, game, source, opponentId);
+
+                        Token citizenOwner = new CitizenGreenWhiteToken();
+                        citizenOwner.putOntoBattlefield(1, game, source, opponentId);
+
+                        break;
+                    case "Secret": // You and that player each draw a card.
+                        Effect drawEffectOpponent = new DrawCardTargetEffect(1);
+                        drawEffectOpponent.setTargetPointer(new FixedTarget(opponentId));
+                        drawEffectOpponent.apply(game, source);
+
+                        Effect drawEffectController = new DrawCardTargetEffect(1);
+                        drawEffectController.setTargetPointer(new FixedTarget(controller.getId()));
+                        drawEffectController.apply(game, source);
+
+                        break;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    private MasterOfCeremoniesChoiceEffect(final MasterOfCeremoniesChoiceEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public MasterOfCeremoniesChoiceEffect copy() {
+        return new MasterOfCeremoniesChoiceEffect(this);
+    }
+}
+
+class MasterOfCeremoniesVote extends VoteHandler<String> {
```

Comment:
unless I’m missing something, does this interact with voting cards like ballot broker? it shouldn’t as it’s not actually voting

Reply:
Nope, you're right. I misread the card as involving a vote rather than a choice. I will have to rework this. That'll be simpler.